lib/CL/pocl_llvm_api.cc

```diff
@@ -668,6 +666,176 @@ static int pocl_llvm_get_kernel_arg_metadata(const char* kernel_name,
   return 0;
 }
 
+
+static int pocl_llvm_get_kernel_arg_metadata_llvm3_9(const char* kernel_name,
```

Comment:
No need to have llvm3_9 in the name. You should select the whole implementation with the macro, thus only have one visible?